crates/nu-command/src/network/http/client.rs

```diff
@@ -221,132 +221,168 @@ pub fn send_request(
+            }
 
-                    let (boundary, data) = (builder.boundary, builder.data);
-                    let content_type = format!("multipart/form-data; boundary={}", boundary);
+            let data = vals
+                .chunks(2)
+                .map(|it| Ok((it[0].coerce_string()?, it[1].coerce_string()?)))
+                .collect::<Result<Vec<(String, String)>, ShellErrorOrRequestError>>()?;
 
-                    let request_fn =
-                        move || req.set("Content-Type", &content_type).send_bytes(&data);
+            let request_fn = map_to_str(data, req);
+            send_cancellable_request(request_url, request_fn, span, signals)
+        }
+        Value::Record { val, .. } => {
+            let mut data: Vec<(String, String)> = Vec::with_capacity(val.len());
 
-                    send_cancellable_request(&request_url, Box::new(request_fn), span, signals)
-                }
-                Value::List { vals, .. } if body_type == BodyType::Form => {
-                    if vals.len() % 2 != 0 {
-                        return Err(ShellErrorOrRequestError::ShellError(ShellError::IOError {
-                            msg: "unsupported body input".into(),
-                        }));
-                    }
+            for (col, val) in val.into_owned() {
+                data.push((col, val.coerce_into_string()?))
+            }
 
-                    let data = vals
-                        .chunks(2)
-                        .map(|it| Ok((it[0].coerce_string()?, it[1].coerce_string()?)))
-                        .collect::<Result<Vec<(String, String)>, ShellErrorOrRequestError>>()?;
-
-                    let request_fn = move || {
-                        // coerce `data` into a shape that send_form() is happy with
-                        let data = data
-                            .iter()
-                            .map(|(a, b)| (a.as_str(), b.as_str()))
-                            .collect::<Vec<(&str, &str)>>();
-                        req.send_form(&data)
-                    };
-                    send_cancellable_request(&request_url, Box::new(request_fn), span, signals)
-                }
-                Value::List { .. } if body_type == BodyType::Json => {
-                    let data = value_to_json_value(&body)?;
-                    send_cancellable_request(
-                        &request_url,
-                        Box::new(|| req.send_json(data)),
-                        span,
-                        signals,
-                    )
+            let request_fn = map_to_str(data, req);
+            send_cancellable_request(request_url, request_fn, span, signals)
+        }
+        _ => Err(ShellErrorOrRequestError::ShellError(ShellError::IOError {
+            msg: "unsupported body input for Form body type".into(),
+        })),
+    }
+}
+
+fn map_to_str(
+    data: Vec<(String, String)>,
+    req: Request,
+) -> Box<impl FnOnce() -> Result<Response, Error>> {
```

Comment:
That naming is not really informative.

Can this be folded into the `send_form_request` handler?

Reply:
I had extracted it since it was used in 2 places, is there a way to do this without duplicating the closure?